### ISSUE TITLE
exported: report differently a method that is now exported

### ISSUE THREAD

**Issue by ccoVeille** [mgechev/revive#1235]
```go
package sandbox

// Foo is OK
func Foo() {}

// bar was not exported but is now exported
func Bar() {}
```

![Image](https://github.com/user-attachments/assets/37143ba2-ee31-4624-bb57-46e401aeeccb)

revive reports the right things yet, but it could be improved.

Here we are 100% sure the issue is the fact the method is now exported, but wasn't before

But because golangci-lint exclude [`EXC0012`](https://golangci-lint.run/usage/false-positives/#exc0012) by default.
This one is not reported via golangci-lint.

We should consider trying to detect it, and report another message that won't be caught by `EXC0012`

Here the idea is to compare that only the case of the first letter differ.

**Comment by chavacava** [comment]
@ccoVeille what message could we report for these cases?

**Comment by ccoVeille** [comment]
Let's consider #1236 with this issue for providing a wording.

So we already have this error messages

- exported (.+) should have comment( \(or a comment on this block\))? or be unexported

- comment on exported (.+) should be of the form "(.+)..."

- package comment should be of the form "(.+)..."

- should have a package comment

These are respectively, exc0012, exc0013, exc0014, exc0015

Here are some suggestions:

- invalid identifier in comment: expected 'FooBarQuux', not 'fooBarQuux'.
- invalid identifier in comment: expected 'FooBarQuux', got 'fooBarQuux'.
- comment should start with 'FooBarQuux', not 'fooBarQuux'
- comment should start with 'FooBarQuux', got 'fooBarQuux'

**Comment by ccoVeille** [comment]
The main reason I wanted to work on this was the fact golangci-lint v1 was excluding almost all errors reported by the `exported` rule within revive.

It was difficult for someone to know that they had to "exclude the exclusions" via `issues.include" filed by using the obscure and cryptic EXC0012-15. Everything was documented, but it wasn't easy. 

Now golangci-lint v2 is here, thanks to the huge amount of work of @ldez and some other people like @alexandear (revive maintainer)

golangci-lint v2 cakes with a lot of changes. One of them is the concept of presets

Now, people who doesn't care about having clean godoc comments will see they are excluded.

Of course, people might have migrated blindly their golangci-lint configuration, so they do have the comments preset, right now.

But I would say, it's OK. They can see it if they look at their configuration then documentation.

TL; DR; I'm unsure we need this refactoring anymore.

So here, is the point do you guys think there is a need to report differently than now issues when:

- a method is now exported or unexported, and the message could be more explicit.
- a method name was changed slightly such as FooJson became FooJSON

Or are we fine with existing reported messages

- comment on exported FooBar should be of the form "FooBar"
- comment on unexported fooBar should be of the form "fooBar ..."
- comment on exported FooJSON should be of the form "FooJSON ..."

This question is for revive maintainers, but it might have consequences on golangci-lint. Because then either they consider it's an interesting use case to report, or if something that will end in "comments" preset (so here it would require extra work in golangci-lint to exclude them)

Maybe we will agree there is some interest in such a thing, maybe not. I'm fine with both. But I'm more inclined to say we can continue with existing code, and do no change at all.

**Comment by ccoVeille** [comment]
I didn't get a feedback about my previous post. Maybe it my fault as it was way too long to read 😅

@chavacava @denisvmedia @alexandear unless give me a feedbacks in the next week, I would consider the issue can be closed with "unplanned" status

**Comment by ccoVeille** [comment]
OK then 👍

I might take a look at it in the next week, but not this weekend.

**Comment by ccoVeille** [comment]
Wording is a pain.

So let's talk here

Here are some suggestions.

Please note I used the [suggestion](https://github.com/mgechev/revive/pull/1403#discussion_r2154852114) made by @alexandear to keep the existing message and only append information to them. 

As a reminder here is the current error messages reported

 - comment on exported function SendJSON should be of the form "SendJSON ..."
 - comment on exported method StatusHTTP.Qux should be of the form "Qux ..."

Suggestion 1:

- comment on exported function SendJSON should be of the form "SendJSON ..." by using its correct casing, not "SendJson".
- comment on exported method StatusHTTP.Qux should be of the form "Qux ..." to match its exported status, not "qux".


Suggestion 2:

- comment on exported function SendJSON should be of the form "SendJSON ...", not "SendJson": casing differs.
- comment on exported method StatusHTTP.Qux should be of the form "Qux ...", not "qux": exported status differs.

Suggestion 3: replace a colon by a comma

- comment on exported function SendJSON should be of the form "SendJSON ...", not "SendJson", casing differs.
- comment on exported method StatusHTTP.Qux should be of the form "Qux ...", not "qux", exported status differs.

Suggestion 4:  variation if the 2nd suggestion without the current value

- comment on exported function SendJSON should be of the form "SendJSON ...": casing differs.
- comment on exported method StatusHTTP.Qux should be of the form "Qux ...": exported status differs.

**Comment by ccoVeille** [comment]
I'm more inclined to choose the 1st suggestion if we are fine my explaining clearly

And suggesting the last one if we are looking for consciousness.

**Comment by ccoVeille** [comment]
About first suggestion

Suggestion 1:

- comment on exported function SendJSON should be of the form "SendJSON ..." by using its correct casing, not "SendJson".
- comment on exported method StatusHTTP.Qux should be of the form "Qux ..." to match its exported status, not "qux".

Please note this wording is the results of a lot of interactions I made by chatting with Google Gemini to help me to write something valid and actionable.

Here are some explanations

<details>
<summary>
It suggested me to use "casing", and not "case"
</summary>
For your linter messages, "casing differs" is significantly better and more accurate than "case differs."
 * It directly addresses the concept of naming conventions and capitalization styles within code, which is the exact problem you're flagging.
 * It's a standard term used in programming contexts (e.g., "case-preserving search," "different casing conventions").
</details>

<details>
<summary>
About the fact to use "by using" (by verb-ing) for one and "to match" (to verb) for the other
</summary>
 * Casing Correction (SendJson -> SendJSON)
   * "by -ing" (by using): This construction feels very natural here. It answers "how should it be?" or "what action should you take?". "By using X" is a direct and common way to instruct someone on the method to achieve something.
   * "to verb" (e.g., to use its correct casing):
     * ... to use its correct casing, not "SendJson".
     * This also works, but "to use" can sometimes imply purpose ("in order to use..."). While it's understandable, "by using" feels slightly more like the direct means of correction.
 * Exported Status Mismatch (qux -> Qux where Qux is now exported)
   * "to verb" (to match): This is very strong here. "To match" directly speaks to the purpose of the change – aligning the comment with the new reality of the method's export status. It's concise and clear about the goal of the correction.
   * "by -ing" (e.g., by matching its exported status):
     * ... by matching its exported status, not "qux".
     * This also works, but "to match" feels slightly more direct for expressing the intent or required outcome when referencing a status. "By matching" sounds a little more like the process rather than the reason/goal.

Conclusion and Recommendation
Based on the natural flow, clarity, and directness, your current suggestion actually hits the sweet spot for both types of messages.
 * For casing corrections, "by using its correct casing" explains the method of correction.
 * For exported status alignment, "to match its exported status" clearly states the goal or reason for the correction.
This combination offers the best clarity and actionability for each specific error type.
</summary>

**Comment by alexandear** [comment]
Thanks for the detailed breakdown and all the work you've put into this, @ccoVeille - especially on clarifying the motivation and the practical implications related to golangci-lint.

I agree with your suggestions, we can implement them if no one else against them.

I think "Suggestion 1" strikes the best balance between clarity and usefulness.

**Comment by ccoVeille** [comment]
> Thanks for the detailed breakdown and all the work you've put into this, @ccoVeille - especially on clarifying the motivation and the practical implications related to golangci-lint.
> 
> I agree with your suggestions, we can implement them if no one else against them.
> 
> I think "Suggestion 1" strikes the best balance between clarity and usefulness.

Please review the draft PR then

https://github.com/mgechev/revive/pull/1403

It's implemented at one place, if it's OK. I'll implement the others

**Comment by ccoVeille** [comment]
Everything is now implemented in the #1403 PR, and ready to be reviewed.

- #1403
